### PR TITLE
Refactor UI and Utils related logic

### PR DESCRIPTION
**Pull request by EgorkaKulikov**

### CODE REVIEW

File: clion-plugin/src/main/kotlin/org/utbot/cpp/clion/plugin/client/Client.kt [hunk 1]
```diff
@@ -192,14 +192,16 @@ class Client(
...
+                val hasChildJobs = requestsCS.coroutineContext.job.children.any()
+                while (hasChildJobs) {
                     delay(DELAY_TIME)
                 }
             }
         }
     }
 
     companion object {
+        //TODO: why don't use process this setting similarly to "isFirstLaunch"? Why is it a property of the client?
```
Comment (vol0n) [inline clion-plugin/src/main/kotlin/org/utbot/cpp/clion/plugin/client/Client.kt:204]: There is no need to store this setting in file system (as we do for `isFirstLaunch`). This setting is changed in tests before starting the plugin for tests. Then when plugin launches, we check this setting in `StartupActivity`. A better a way would be not use this "global" variable, but ask from IDE whether it was launched in test mode or not

File: clion-plugin/src/main/kotlin/org/utbot/cpp/clion/plugin/utils/ClientUtils.kt [hunk 2]
```diff
@@ -2,36 +2,24 @@ package org.utbot.cpp.clion.plugin.utils
...
 import com.intellij.openapi.project.Project
-import kotlinx.coroutines.CoroutineScope
-import kotlinx.coroutines.job
 import org.utbot.cpp.clion.plugin.client.Client
 import org.utbot.cpp.clion.plugin.client.ClientManager
 import org.utbot.cpp.clion.plugin.client.logger.ClientLogger
 
-val Project.logger: ClientLogger
-    get() = this.service()
+//TODO: remove logger from this file
```
Comment (vol0n) [inline clion-plugin/src/main/kotlin/org/utbot/cpp/clion/plugin/utils/ClientUtils.kt:11]: maybe you have some comments why it should be removed or some ideas where it should be. The logger is accessed from different places (requests, handlers, some UI components) to show logs to the user

File: clion-plugin/src/main/kotlin/org/utbot/cpp/clion/plugin/client/Client.kt (hunk 1 above)
Comment (vol0n) [reply clion-plugin/src/main/kotlin/org/utbot/cpp/clion/plugin/client/Client.kt:204]: I found `Application.isUnitTestMode`, so this variable is not needed. You can remove it, or leave this todo and I will fix that

**Review by vol0n — CHANGES_REQUESTED** [review]

File: clion-plugin/src/main/kotlin/org/utbot/cpp/clion/plugin/client/Client.kt (hunk 1 above)
Comment (EgorkaKulikov) [reply clion-plugin/src/main/kotlin/org/utbot/cpp/clion/plugin/client/Client.kt:204]: Ok. I removed TODO.

File: clion-plugin/src/main/kotlin/org/utbot/cpp/clion/plugin/utils/ClientUtils.kt (hunk 2 above)
Comment (EgorkaKulikov) [reply clion-plugin/src/main/kotlin/org/utbot/cpp/clion/plugin/utils/ClientUtils.kt:11]: I hope that we will be able to implement logger without services similarly to UtBot Java approach

File: clion-plugin/src/main/kotlin/org/utbot/cpp/clion/plugin/client/Client.kt [hunk 3]
```diff
@@ -192,14 +192,16 @@ class Client(
     fun waitForServerRequestsToFinish(timeout: Long = SERVER_TIMEOUT) {
         runBlocking {
             withTimeout(timeout) {
-                while (requestsCS.hasChildren()) {
+                val hasChildJobs = requestsCS.coroutineContext.job.children.any()
+                while (hasChildJobs) {
```
[the remark this answers is not in the document]
Comment (EgorkaKulikov) [reply clion-plugin/src/main/kotlin/org/utbot/cpp/clion/plugin/client/Client.kt]: Thanks! Fixed

**Review by vol0n — APPROVED** [review]

**Pull request merged**